### ISSUE TITLE
v-on="click: property = !property"

### ISSUE THREAD

**Issue by aristidesfl** [vuejs/Discussion#124]
```
<button v-on="click: showDetails = !showDetails">Show details</button>
<span v-if="showDetails">Details</span>
```
http://jsbin.com/nikuqeyesi/2/edit

This pattern is quite useful but doesn't work unless the property is initialised explicitly in `vm.$data`. It would be nice not having to pollute the vm instantiation code and data option with presentation only properties, like it is already possible with inputs which use v-model. 

Also with v-model, you can set your initial value with `value=""`. Maybe something like this could be used to set the initial value of `showDetails`?

Is this possible to achieve already in a simple way? Maybe a special filter like `key`, or a directive, or a way to specify the initial value like `value=""`?

**Comment by yyx990803** [comment]
There are different ways to approach this - I usually put the actual "model" data in a separate object, and allow presentation state properties in the top level. Or you can do the opposite, keep all presentation properties in a nested object. The case for `v-model` is because it's two way so it performs the write even without the initial value, but it's not recommended to rely on that.

**Comment by aristidesfl** [comment]
I see, so a `v-toggle="data"` directive which toggles the model.property on click isn't justified just for this case?

**Comment by yyx990803** [comment]
Yeah, I don't think `v-toggle` should be a built-in directive, but you can always use it as a custom one. The label usage is clever :)

**Comment by hlooalklampc** [comment]
<script>
          new Vue({
        el: '#bl',
        data: { showDetails: true }
    })
  </script>
http://jsfiddle.net/wvr0yx5h/4/
